scss/forms/_floating-labels.scss

```diff
@@ -0,0 +1,65 @@
+// stylelint-disable selector-no-vendor-prefix
+
+.form-floating {
+  position: relative;
+
+  > label,
+  > .form-control,
+  > .form-select {
+    height: $form-floating-height;
+    padding: $form-floating-padding-y $form-floating-padding-x;
+  }
+
+  > label {
+    position: absolute;
+    top: 0;
+    left: 0;
+    display: block;
+    width: 100%;
+    margin-bottom: 0; // Override default `<label>` margin
+    pointer-events: none;
+    cursor: text; // Match the input under the label
+    border: $input-border-width solid transparent;
+    @include border-radius(.25rem);
+    @include transition(all .1s ease-in-out);
```

Comment:
I would explicitly set `opacity` and `transform` here, like we do with other elements (eg. buttons)